packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/javamodel/JavaPackage.java

```diff
@@ -308,7 +308,7 @@ public void addModelUnitTest(ClientModel model) {
             JavaFile javaFile = javaFileFactory.createTestFile(packageName, className);
             ModelTestTemplate.getInstance().write(model, javaFile);
             if (!this.checkDuplicateFile(javaFile.getFilePath())) {
-                javaFiles.add(javaFile);
+                addJavaFile(javaFile);
```

Comment:
The problem is code used `javaFIles.add`.

Only `addJavaFile` method would log the file names.